ynr/apps/people/helpers.py

```diff
@@ -120,6 +120,30 @@ def clean_twitter_username(username):
     return username
 
 
+def clean_instagram_url(url):
+    parsed_username = urlparse(url)
+    if parsed_username.netloc not in [
+        "instagram.com",
+        "www.instagram.com",
+        "instagr.am",
+        "www.instagr.am",
+    ]:
+        raise ValueError(
+            "The Instagram URL must be from a valid Instagram domain."
+        )
+    username = parsed_username.path.strip("/")
+    if not re.search(
+        r"^([A-Za-z0-9_](?:(?:[A-Za-z0-9_]|(?:\.(?!\.))){0,28}(?:[A-Za-z0-9_]))?)$",
+        username,
+    ):
+        raise ValueError(
+            "This is not a valid Instagram username. Please try again."
+        )
+    return "https://{domain}/{username}".format(
+        domain=parsed_username.netloc, username=username
```

Comment:
As `www.instagram.com` is the canonical URL, I think we can hard code that here. `instagr.am` and the other domains redirect there anyway, so we may as well anticipate that here